src/js/components/award/shared/activity/ContractGrantsActivityChart.jsx

```diff
@@ -97,23 +121,117 @@ const ContractGrantsActivityChart = ({
+        return { date: newDate, label };
+    });
+    /**
+     * createXScaleAndTicks
      * - creates the x scaling function and updates state
      * @returns {null}
      */
-    // const createXScaleAndTicks = useCallback(() => {
-    //     const scale = scaleLinear().domain(xDomain).range([0, visualizationWidth]).nice();
-    //     const ticks = scale.ticks(6);
-    //     // add last tick for spacing
-    //     const updatedTicksWithSpacing = addTicksForSpacing(ticks, true);
-    //     // create new scale since we have new data
-    //     const updatedScale = scaleLinear()
-    //         .domain([xDomain[0], updatedTicksWithSpacing[updatedTicksWithSpacing.length - 1]])
-    //         .range([0, visualizationWidth])
-    //         .nice();
-    //     setXTicks(updatedTicksWithSpacing);
-    //     setXScale(() => updatedScale);
-    // }, [xDomain, visualizationWidth]);
+    const createXScaleAndTicks = useCallback(() => {
+        if (!xDomain.length) return null;
+        /**
+         * By design, we want to leave space between the start of the chart
+         * and the end of the chart.
+         */
+        const spacing = visualizationWidth * 0.05;
+        const scale = scaleLinear()
+            .domain(xDomain)
+            .range([spacing, visualizationWidth - spacing - padding.left]);
+        let theTicks = scale.ticks(6);
+        // add first and last tic per design
+        theTicks.splice(0, 0, xDomain[0]); // first tick
+        theTicks.push(xDomain[1]); // last tick
+
+        const differences = theTicks.reverse().reduce((acc, tick, i, src) => {
+            const nextTick = src[i + 1];
+            if (!nextTick) return acc; // last tick ignore
+            acc.push(tick - nextTick);
+            return acc;
+        }, []);
+
+        const averageDifference = sum(differences) / theTicks.length;
+        // ascending order since we reversed above
+        theTicks.reverse();
+        /**
+         * manuallyCreateXTicks
+         * - Manually create the x ticks starting with the first tick and
+         * adding the difference sequentially.
+         */
+        const manuallyCreateXTicks = (ticks, diff) => {
+            const newTicks = cloneDeep(ticks);
+            /**
+             * Tthe ticks method for d3 does not include ticks for the very beginning
+             * and the end of the domain. This leaves a gap between the last tick we create
+             * and the last tick of the domain. Therefore we increase our array by two.
+             * Since we ignore the first and last tick we cannot unshift or push :(.
+             */
+            newTicks.splice(1, 0, 1);
+            return newTicks.reduce((acc, tick, i) => {
+                // ignore first and last tick
+                if (i === 0 || i + 1 === newTicks.length) {
+                    acc.push(tick);
+                }
+                else {
+                    acc.push(acc[i - 1] + diff);
```

Comment:
Why do we do this? Can we rename this function and the const `newTicks` to reflect the purpose of the function and the resulting return value?

IE

`addOffsetToTicks = () =>`

and

`ticksWithOffset`

Reply:
This generates new ticks that have equal spacing using the average difference. We could rename it `evenlySpacedTicks`?

Reply:
Yes. Follow up PR?

Reply:
done in the last PR